ZProbe porting, temp fix for sleep conflicts, start of DistanceProbe module

DouglasPearless opened the pull request:
This pull request includes:
(1) ZProbe port from current Edge on Smoothie1
(2) Update to sleep in mbed to allow compiling, note this should probably be fixed via a preprocessor macro?
(3) The start of the DistanceProbe module, this is incomplete and therefore does not yet work
(4) Update to checksumm.h for Eclipse (can we have this in the master branch as it seems to work OK in Eclipse and command line make
(5) A makefile in the Smoothie2 directory that was ported from Smoothie1

DouglasPearless committed:
modified:   gcc4mbed/external/mbed/libraries/mbed/hal/sleep_api.h
	new file:   makefile
	modified:   src/config.default
	modified:   src/libs/Kernel.h
	modified:   src/libs/checksumm.h
	modified:   src/main.cpp
	modified:   src/makefile
	modified:   src/modules/robot/Conveyor.cpp
	modified:   src/modules/robot/Conveyor.h
	new file:   src/modules/tools/zprobe/DeltaCalibrationStrategy.cpp
	new file:   src/modules/tools/zprobe/DeltaGridStrategy.cpp
	new file:   src/modules/tools/zprobe/ZGridStrategy.cpp
	new file:   src/modules/tools/zprobe/ZProbe.cpp
	new file:   src/modules/tools/zprobe/ZProbe.h
	src/config.default_backup

DouglasPearless committed:
new file:   src/modules/tools/distanceprobes/DistanceProbes.cpp
	new file:   src/modules/tools/distanceprobes/DistanceProbes.h
	new file:   src/modules/tools/distanceprobes/DistanceProbesPublicAccess.h
	new file:   src/modules/tools/distanceprobes/DistanceSensor.h
	new file:   src/modules/tools/distanceprobes/InfraredProbe.cpp
	new file:   src/modules/tools/distanceprobes/InfraredProbe.h
	new file:   src/modules/tools/distanceprobes/VCNL40x0.cpp
	new file:   src/modules/tools/distanceprobes/VCNL40x0.h
	new file:   src/modules/tools/zprobe/DeltaCalibrationStrategy.h
	new file:   src/modules/tools/zprobe/DeltaGridStrategy.h
	new file:   src/modules/tools/zprobe/LevelingStrategy.h
	new file:   src/modules/tools/zprobe/Plane3D.cpp
	new file:   src/modules/tools/zprobe/Plane3D.h
	new file:   src/modules/tools/zprobe/ThreePointStrategy.cpp
	new file:   src/modules/tools/zprobe/ThreePointStrategy.h
	new file:   src/modules/tools/zprobe/ZGridStrategy.h

DouglasPearless committed:
modified:   src/libs/Kernel.h

DouglasPearless committed:
modified:   src/config.default
	modified:   src/libs/Kernel.h

adamgreen commented on src/libs/checksumm.h:
```diff
@@ -1,36 +1,36 @@
-    You should have received a copy of the GNU General Public License along
-    with Smoothie. If not, see <http://www.gnu.org/licenses/>.
-*/
+ This file is part of Smoothie (http://smoothieware.org/)
```
Why has the formatting been changed so much for this file? The diff is so big that I can't even see what was actually changed in it. This should be fixed.

DouglasPearless replied:
Yes, very good question, it wasn't something I consciously did, will manually fix.

adamgreen commented on gcc4mbed/external/mbed/libraries/mbed/hal/sleep_api.h:
```diff
@@ -18,7 +18,9 @@
 #include "device.h"
 
-#if DEVICE_SLEEP
+//#if DEVICE_SLEEP //TODO THIS WAS THE ORIGINAL BUT CONFLICTS WITH OTHER DEFINITION OF SLEEP
```
Can you describe the problem that you were seeing which caused this change to be required? We should not be checking in anything to this part of the tree since it will be lost the next time we update GCC4MBED and the mbed SDK.

DouglasPearless replied:
SiLabs (aka Energy Micro) have started to port their energy management code to MBED and this causes conflicts with the definition of sleep.  If you do not make this (temporary change) then Smoothie2 will no longer compiles.  Someone more knowledgeable than me with preprocessors could probably workout an override to undef DEVICE_SLEEP

adamgreen commented on makefile:
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/make
```
I went through all that work in GCC4MBED to get rid of recursive makefiles and now we go and add one back? If we really want the makefile to be in the root directory then it is probably better to just move the one currently in src/ and update a few things to let it know that the src can be found in src/ instead of ./

DouglasPearless replied:
Happy to drop this, I simply noted it was in Smoothie1 and not Smoothie2 so ported it.

adamgreen commented on src/makefile:
```diff
@@ -29,8 +29,6 @@ MRI_SEMIHOST_STDIO  := 0
 # Use C++11 features for the checksums
 DEFINES += -DCHECKSUM_USE_CPP
 
-
```
This doesn't seem all that useful.

DouglasPearless replied:
True, just unnecessary white space removed.

adamgreen commented on src/modules/tools/zprobe/DeltaGridStrategy.cpp:
```diff
@@ -0,0 +1,600 @@
+/*
+ This code is derived from (and mostly copied from) Johann Rocholls code at https://github.com/jcrocholl/Marlin/blob/deltabot/Marlin/Marlin_main.cpp
+ license is the same as his code.
```
Maybe copying in that copyright message instead of making a reader go and find it would be better? What license are the other new files in this PR being added with? Some don't currently have any license explicitly stated.

DouglasPearless replied:
Ah, I just used the Eclipse create .cpp and .h function, I will copy the Smoothie license into each of the files.

adamgreen commented on src/modules/robot/Conveyor.cpp:
```diff
@@ -58,6 +58,45 @@ Conveyor::Conveyor(){
+{
+    if(queue.is_empty()) {
+
+//    	for (size_t i = 0; i < THEKERNEL->robot->actuators.size(); i++) {
```
In general I don't like commits that include a bunch of commented out code. It makes the code harder to read and also tends to make the code looked hacked upon until something finally worked. I would prefer to see a commit added to the PR which just deletes such commented lines out of the code so that they only exist in the git history. @arthurwolf might feel differently though.

DouglasPearless replied:
I will reverted back to Smoothie2 version

DouglasPearless committed:
modified:   src/libs/checksumm.h

DouglasPearless committed:
modified:   src/libs/checksumm.h

DouglasPearless committed:
modified:   src/libs/checksumm.h

DouglasPearless committed:
modified:   src/makefile

DouglasPearless committed:
modified:   src/modules/tools/distanceprobes/DistanceProbes.cpp
	modified:   src/modules/tools/distanceprobes/DistanceProbes.h
	modified:   src/modules/tools/distanceprobes/DistanceProbesPublicAccess.h
	modified:   src/modules/tools/distanceprobes/DistanceSensor.h
	modified:   src/modules/tools/distanceprobes/InfraredProbe.cpp
	modified:   src/modules/tools/distanceprobes/InfraredProbe.h

wolfmanjm commented:
This can't possibly work. The zprobe stuff relies on the new motion control and that has not been ported, stepticker, conveyor etc etc all need to be updated for this to work at all. This should most definitely not be merged until the actual port of the new motion control has been done.

DouglasPearless commented:
OK, While I was aware that there is a lot do, I was not aware of all of that as such, I will continue my porting of stepstick, conveyor [and what else?]  Cheers Douglas

wolfmanjm commented:
as I said virtually every file changed, robot.cpp, steppermotor.cpp.
I think unless the entire new motion control is ported first this hybrid zprobe stuff needs to go.
I think if you insist on doing this now you should port the zprobe stuff from the master branch not the current edge. That will work with the old motion control and probably not need any changes to work. What you have now simply will not work.

DouglasPearless commented:
OK, Then I misunderstood what was required to get ZProbe working in Smoothie2, so if I understand correctly, I have two options:
(1) if I throw away my changes, re-port the ZProbe from the Smoothie1 master and not edge to use the ‘old’ motion control which I assume is still in Smoothie2?
(2) Continue to try and port all the motion control stuff to V2 and get it working
Then I can work on my implementation of the RotaryDelta calibration routines (my end goal).

Cheers
Douglas
> On 12/09/2016, at 3:28 PM, Jim Morris <notifications@github.com> wrote:
> 
> as I said virtually every file changed, robot.cpp, steppermotor.cpp.
> I think unless the entire new motion control is ported first this hybrid zprobe stuff needs to go.
> I think if you insist on doing this now you should port the zprobe stuff from the master branch not the current edge. That will work with the old motion control and probably not need any changes to work. What you have now simply will not work.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/Smoothieware/Smoothie2/pull/29#issuecomment-246235549>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ADkh-8KvbP29flkRR46cenISHyErbCQFks5qpMbTgaJpZM4J6KZv>.
>

DouglasPearless committed:
deleted:    makefile

wolfmanjm commented:
Sorry if we were unclear... Your two options are...

1. port the zprobe stuff from master which will match the current smoothie2 motion control and will need virtually no changes. it is also fully featured as the new edge zprobe stuff is.

2. Upgrade smoothie2 codebase to the new motion control which basically touched virtually ever file, then the new edge zprobe stuff will just work and will also need no modifications.

What will not work is using the current edge zprobe code on the old motion control platform as you were trying to do.

DouglasPearless committed:
modified:   ../gcc4mbed/external/mbed/libraries/mbed/hal/sleep_api.h
	new file:   libs/MemoryPool.cpp
	new file:   libs/MemoryPool.h
	modified:   libs/checksumm.h
	new file:   libs/platform_memory.cpp
	new file:   libs/platform_memory.h
	modified:   main.cpp
	modified:   modules/robot/Conveyor.cpp
	modified:   modules/robot/Conveyor.h
	deleted:    modules/tools/distanceprobes/DistanceProbes.cpp
	deleted:    modules/tools/distanceprobes/DistanceProbes.h
	deleted:    modules/tools/distanceprobes/DistanceProbesPublicAccess.h
	deleted:    modules/tools/distanceprobes/DistanceSensor.h
	deleted:    modules/tools/distanceprobes/InfraredProbe.cpp
	deleted:    modules/tools/distanceprobes/InfraredProbe.h
	deleted:    modules/tools/distanceprobes/VCNL40x0.cpp
	deleted:    modules/tools/distanceprobes/VCNL40x0.h
	modified:   modules/tools/zprobe/DeltaCalibrationStrategy.cpp
	modified:   modules/tools/zprobe/DeltaGridStrategy.cpp
	modified:   modules/tools/zprobe/ThreePointStrategy.cpp
	modified:   modules/tools/zprobe/ZGridStrategy.cpp
	modified:   modules/tools/zprobe/ZProbe.cpp
	modified:   modules/tools/zprobe/ZProbe.h

DouglasPearless committed:
modified:   libs/checksumm.h

DouglasPearless committed:
new file:   modules/tools/distanceprobes/DistanceProbes.cpp
	new file:   modules/tools/distanceprobes/DistanceProbes.h
	new file:   modules/tools/distanceprobes/DistanceProbesPublicAccess.h
	new file:   modules/tools/distanceprobes/DistanceSensor.h
	new file:   modules/tools/distanceprobes/InfraredProbe.cpp
	new file:   modules/tools/distanceprobes/InfraredProbe.h
	new file:   modules/tools/distanceprobes/VCNL40x0.cpp
	new file:   modules/tools/distanceprobes/VCNL40x0.h

DouglasPearless commented:
I have implemented:
1. port the zprobe stuff from master which will match the current smoothie2 motion control and will need virtually no changes. it is also fully featured as the new edge zprobe stuff is.

Would it be easier to reject my pull request and for me to submit a new pull request?

arthurwolf commented:
Hey.

Yeah, ideally you want to make a new clean merge request

Cheers.

On Wed, Sep 14, 2016 at 11:50 AM, Douglas Pearless <notifications@github.com
> wrote:

> I have implemented:
> 1. port the zprobe stuff from master which will match the current
> smoothie2 motion control and will need virtually no changes. it is also
> fully featured as the new edge zprobe stuff is.
>
> Would it be easier to reject my pull request and for me to submit a new
> pull request?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Smoothieware/Smoothie2/pull/29#issuecomment-246961993>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGpFWkWd73int5j7PhBunr1_wx4-mmaks5qp8NPgaJpZM4J6KZv>
> .
>



-- 
Courage et bonne humeur.

DouglasPearless committed:
modified:   main.cpp

DouglasPearless commented:
Yes, can you please reject my pull request

arthurwolf commented:
There you go :)
